service/controller/v22/resource/s3bucket/delete.go

```diff
@@ -8,10 +8,22 @@ import (
 	"github.com/aws/aws-sdk-go/service/s3"
 	"github.com/giantswarm/microerror"
 	"github.com/giantswarm/operatorkit/controller"
+	"github.com/giantswarm/operatorkit/controller/context/finalizerskeptcontext"
+	"github.com/giantswarm/operatorkit/controller/context/resourcecanceledcontext"
+	"golang.org/x/sync/errgroup"
 
 	"github.com/giantswarm/aws-operator/service/controller/v22/controllercontext"
 )
 
+const (
+	// loopLimit is the maximum amount of delete actions we want to allow per
+	// reconciliation loop. Reason here is to execute resources fast and prevent
+	// them blocking other resources for too long. In case a S3 bucket has
+	// thousands of objects, we delete 3 times 1000 thousand and continue with the
+	// next reconciliation loop.
```

Comment:
```
In case a S3 bucket has thousands of objects, we delete 3 times 1000 thousand
```

I don't understand this part. I thought in this case we deleted just 3.